src/core/domain/DTOs/patch_info_dto.py

```diff
@@ -0,0 +1,31 @@
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, constr
+from pydantic_factories import ModelFactory
+
+
+class status_enum(str, Enum):
+    active = "active"
+    resolved = "resolved"
+    muted = "muted"
+
+
+class level_enum(str, Enum):
+    critical = "critical"
+    error = "error"
+    warning = "warning"
+    info = "info"
+    debug = "debug"
+
+
+class PatchInfo(BaseModel):
+    status: Optional[status_enum]
+    resolved_in_version: Optional[constr(max_length=40)]  # if status = resolved
+    title: Optional[constr(min_length=1, max_length=255)]
```

Comment:
I think this will fail in mypy.

There is another solution.

You can find it in the `NonEmptyStrictStr` class of `integration` project

Reply:
Oh ok ty. Done!